src/core/vm.c

```diff
@@ -319,6 +318,12 @@ struct vm* vm_init(struct vm_allocation* vm_alloc, const struct vm_config* vm_co
 
     cpu_sync_barrier(&vm->sync);
 
+    if (master) {
+        vm_mem_prot_init(vm, vm_config);
+    }
+
+    cpu_sync_barrier(&vm->sync);
+
```

Comment:
I don't understand why is moved from `vm_master_init` to here, at least in this PR